src/components/Dashboard.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+    <div class="h-full w-full flex justify-center items-center bg-[#141e36] flex-col">
+        <h1 class="text-5xl bg-transparent text-white mb-8">Dashboard</h1>
+        <h2 class="text-3xl text-white italic mb-4">Total amount: {{ totalAmount }}</h2>
+        <h3 class="text-2xl text-white italic">Transaktonen</h3>
+        <div class="h-3/4 py-12 w-1/2">
+            <div v-for="transaction in transactions" class="p-4 flex justify-between w-full" v-bind:key="transaction.id">
+                <span class="text-white">{{ transaction.title }}</span>
+                <span class="text-white">{{ transaction.positiv ? "+" : "-" }}{{ transaction.amount }}</span>
+            </div>
+            <div class="mt-4 flex justify-center items-center w-full h-10 border rounded-lg border-dashed cursor-pointer" @click="addTransaction">
+                <span class="text-white text-2xl">+</span>
+            </div>
+        </div>
+    </div>
+</template>
+<script lang="ts">
+import type { Transaction } from '@/models/transaction';
+import { getTransactions } from '@/service/data';
+
+export default {
+    name: "InsimodusDashboard",
+    data(){
+        return {
+            transactions: [] as Transaction[]
+        }
+    },
+    beforeMount() {
+        this.transactions = getTransactions();
+    },
+    computed: {
+        totalAmount(){
+            let sum = 0;
+            this.transactions.map(transaction => sum += transaction.positiv ? transaction.amount : -transaction.amount)
+
+            return sum
+        }
+    },
```

Comment:
Warum wird hier das API-Modul computed verwendet? Man könnte totalAmount ja in data() definieren und eine Methode dazuschreiben?

Reply:
Wenn man Methoden in einem Template verwendet, werden sie bei jedem Render-Zyklus ausgeführt. Das ist oft unproblematisch, kann jedoch je nach Komplexität der Methode oder des Templates zu erheblichen Performanceeinbußen führen.

VueJS stellt aus diesem Grund sogenannte computed-Methoden bereit. Diese haben die Eigenschaft, dass sie nur dann erneut ausgeführt werden, wenn sich ihre Eingabewerte ändern. Natürlich könnte man den totalAmount auch als Datenobjekt definieren und bei jedem Neuladen der Transaktionen neu berechnen. Allerdings bietet computed genau diese Funktionalität automatisch an und erlaubt es dabei, alles an einem zentralen Punkt zu definieren, was die Lesbarkeit des Codes erheblich verbessert.

https://vuejs.org/guide/essentials/computed.html

Reply:
Sehr einleuchtend, danke für die Erklärung!